components/blog/PostList.tsx

```diff
@@ -0,0 +1,124 @@
+import Image from "next/image";
+import Link from "next/link";
+import { getStrapiMedia, formatDate } from "../../utils/blog/api-helpers";
+
+export interface Article {
+	id: number;
+	attributes: {
+		title: string;
+		description: string;
+		slug: string;
+		createdAt: string;
+		updatedAt: string;
+		publishedAt: string;
+		cover: {
+			data: {
+				attributes: {
+					url: string;
+				};
+			};
+		};
+		category: {
+			data: {
+				attributes: {
+					name: string;
+					slug: string;
+				};
+			};
+		};
+		authorsBio: {
+			data: {
+				attributes: {
+					name: string;
+					avatar: {
+						data: {
+							attributes: {
+								url: string;
+							};
+						};
+					};
+				};
+			};
+		};
+	};
+}
+
+interface PostListProps {
+	data: Article[];
+	children?: React.ReactNode;
+}
+const PostList = ({
+	data: articles,
+	children,
+}: PostListProps) => {
+	return (
+		<section className="container p-6 mx-auto space-y-6 sm:space-y-12">
+			<div className="grid justify-center grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3">
+				{articles.map((article) => {
+					const imageUrl = getStrapiMedia(
+						article.attributes.cover.data?.attributes.url
+					);
+
+					const category = article.attributes.category.data?.attributes;
+					const authorsBio = article.attributes.authorsBio.data?.attributes;
+
+					const avatarUrl = getStrapiMedia(
+						authorsBio?.avatar.data.attributes.url
+					);
+					if (imageUrl === null || avatarUrl === null) {
+						console.error(`[PostList] imageUrl = ${imageUrl}, avatarURL = ${avatarUrl}`);
+						return <></>;
```

Comment:
Yahan bhi, we are not showing the article at all if the images are not found. 
Maybe, the default fallback images can be part of the same module that serves the `getStrapiMedia` function.

And you can use that here as well.

Reply:
You can turn this into an issue if you don't want to handle this right away.

Reply:
This check wasn't really needed as imageUrl and avatarUrl are being checked in the return statement too. Removed it